firestore/useCollection.ts

```diff
@@ -123,7 +124,7 @@ const useCollectionDataInternal = <
               transform
             )
           )
-        : undefined) as Data<T, IDField, RefField>[],
+        : options && options.initialValue) as Data<T, IDField, RefField>[],
```

Comment:
If we promote this to v5, I would suggest we change this to `options?.initialValue` as an easier way of safe navigation